Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM nodesource/node:6
+
+ADD https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_amd64 /usr/local/bin/dumb-init
+
+RUN chmod +x /usr/local/bin/dumb-init
+
+RUN groupadd --system -- nodejs && \
+    useradd --system --gid nodejs --create-home -- nodejs
+
+USER nodejs
+
+RUN mkdir -p /home/nodejs/silly-goat
+
+WORKDIR /home/nodejs/silly-goat
+
+COPY package.json .
+
+RUN npm install --production
+
+COPY . .
+
+ENTRYPOINT ["dumb-init", "bin/hubot"]
```

Comment:
> Btw, if you want to set Hubot to listen on a specific port you can do `ENV PORT 3000`.

Does Hubot listen? My understanding is that it makes a request to the Gitter streaming API and consumes the response indefinitely.

> Also, is it possible to make CircleCI build and push new image to DockerHub whenever there's a successful merge to `master`?

Would it also be possible to automatically update the Linode in response to this push? If not it seems reasonable to me to publish to Docker Hub via xyz, then pull and restart the bot on the Linode manually.